.github/scripts/deploy-utils.sh

```
@@ -219,3 +219,155 @@ print_script_info() {
+restore_from_checkpoint() {
+    local deploy_dir="$1"
+    local environment="$2"
+    local checkpoint_file="${deploy_dir}/.deployment-checkpoint"
+
+    if [[ ! -f "$checkpoint_file" ]]; then
+        log_error "No checkpoint file found at: $checkpoint_file"
+        return 1
+    fi
+
+    # Checkpoint 로드
+    source "$checkpoint_file"
```

Shellcheck (0.11.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)